### PR TITLE
feat: Adding Audio processing features

### PR DESCRIPTION
**Pull request by AshAnand34**
# Add Audio Trim and Merge Tools (#161)

### Audio Trim Tool (`/audio/trim`)
- Cut and trim audio files by specifying start and end times
- Support for MP3, AAC, and WAV output formats
- Time input in HH:MM:SS format (e.g., 00:00:30 to 00:01:30)
- Perfect for extracting specific segments from longer audio files

### Audio Merge Tool (`/audio/merge-audio`)
- Combine multiple audio files into a single file
- Drag-and-drop interface for multiple file uploads
- Files are merged in the order they're uploaded
- Support for MP3, AAC, and WAV output formats
- Ideal for combining podcast segments or music tracks

### Audio Speed Change Tool (`/audio/change-speed`)
- Adjust playback speed of audio files (0.5x to 4x)
- Support for MP3, AAC, and WAV output formats
- Maintains audio quality while changing tempo
- Perfect for podcast listening or music production

### Audio Extraction Tool (`/audio/extract-audio`)
- Extract audio tracks from video files
- Support for MP3, AAC, and WAV output formats
- Works with various video formats (MP4, MOV, AVI, etc.)
- Ideal for creating audio-only versions of videos

## 🔧 Technical Implementation

- **FFmpeg Integration**: All tools use FFmpeg.wasm for client-side audio processing
- **Multi-file Input Component**: Created `ToolAudioInput` and `ToolMultipleAudioInput` components for handling 1 and multiple audio files, respectively.
- **Comprehensive Testing**: Added test suites with FFmpeg mocking for Node.js compatibility
- **Type Safety**: Full TypeScript implementation with proper type definitions
- **Client-side Processing**: All audio processing happens in the browser, ensuring privacy

## 🧪 Testing

- All tools include comprehensive test suites
- FFmpeg properly mocked for Node.js test environment
- Tests cover different output formats and edge cases
- Mock implementations ensure tests run reliably in CI/CD

### Audio Trim:
- Podcast editing (removing intro/outro)
- Music production (extracting specific parts)
- Audio cleanup (removing unwanted segments)

### Audio Merge:
- Podcast episode assembly
- Music track compilation
- Audio book chapter combination

### Audio Speed Change:
- Podcast listening at different speeds
- Music production and remixing
- Language learning with slower playback

### Audio Extraction:
- Creating audio-only versions of videos
- Podcast creation from video content
- Music extraction from music videos

## 🚀 Features

- **Privacy-focused**: All processing happens client-side
- **No upload limits**: Process files of any size
- **Multiple formats**: Support for MP3, AAC, and WAV
- **Professional quality**: FFmpeg-powered processing
- **User-friendly**: Intuitive drag-and-drop interface

### CODE REVIEW

File: src/pages/tools/audio/merge-audio/service.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import { FFmpeg } from '@ffmpeg/ffmpeg';
+import { fetchFile } from '@ffmpeg/util';
+import { InitialValuesType } from './types';
+
+const ffmpeg = new FFmpeg();
+
+export async function mergeAudioFiles(
```
Comment (iib0011) [inline src/pages/tools/audio/merge-audio/service.ts:7]: this function only works if the input audio files have the same format. Try to combine aac and mp3

File: src/pages/tools/audio/change-speed/service.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { InitialValuesType } from './types';
+
+export function main(
```
Comment (iib0011) [inline src/pages/tools/audio/change-speed/service.ts:3]: this is used by the test

**Review by iib0011 — CHANGES_REQUESTED** [review]

File: src/pages/tools/audio/change-speed/service.ts (hunk 2 above)
Comment (iib0011) [reply src/pages/tools/audio/change-speed/service.ts:3]: you can move the logic from the index.tsx

**Comment by iib0011** [comment]
Great job.

**Review by iib0011 — APPROVED** [review]
